components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1269,7 +1270,131 @@ where
             SignificantMsg::RaftlogFetched { context, res } => {
                 self.on_raft_log_fetched(context, res);
             }
+            SignificantMsg::EnterForceLeaderState { failed_stores } => {
+                self.on_enter_pre_force_leader(failed_stores)
+            }
+            SignificantMsg::ExitForceLeaderState => self.on_exit_force_leader(),
+        }
+    }
+
+    fn on_enter_pre_force_leader(&mut self, failed_stores: HashSet<u64>) {
+        if self.fsm.peer.force_leader.is_some() {
+            return;
+        }
+
+        let expected_alive_voter = self
+            .fsm
+            .peer
+            .raft_group
+            .raft
+            .prs()
+            .conf()
+            .voters()
+            .ids()
+            .iter()
+            .filter(|v| !failed_stores.contains(v))
```

Comment:
Probably wrong to comparing store id to peer id.

Reply:
oh, my fault. the store id is the same as peer id, so the tests can pass.

Reply:
fixed